src/Spryker/Zed/EventBehavior/Business/Model/EventResourceQueryContainerPluginIterator.php

```diff
@@ -17,25 +17,33 @@ class EventResourceQueryContainerPluginIterator extends AbstractEventResourcePlu
      */
     protected $plugin;
 
+    /**
+     * @var array<int>
+     */
+    protected $ids;
+
     /**
      * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourceQueryContainerPluginInterface $plugin
      * @param int $chunkSize
+     * @param array<int> $ids
      */
-    public function __construct(EventResourceQueryContainerPluginInterface $plugin, int $chunkSize)
+    public function __construct(EventResourceQueryContainerPluginInterface $plugin, int $chunkSize, $ids = [])
     {
         parent::__construct($plugin, $chunkSize);
+
+        $this->ids = $ids;
     }
 
     /**
      * @return void
      */
     protected function updateCurrent(): void
     {
+        $whereCondition = !$this->ids ? ModelCriteria::ISNOTNULL : sprintf('%s (%s)', ModelCriteria::IN, implode(',', $this->ids));
         $this->current = $this->plugin->queryData()
             ->offset($this->offset)
             ->limit($this->chunkSize)
-            ->where($this->plugin->getIdColumnName() . ModelCriteria::ISNOTNULL)
-            ->select([$this->plugin->getIdColumnName()])
```

Comment:
Why did you remove this? doesn't it now load full object?

Reply:
yes. it's not enough for the eventTransfer to have only id